packages/in17n/src/register.ts

```diff
@@ -0,0 +1,69 @@
+import { Awaited, Plugin, preGenericsInitialization, preLogin, SapphireClient } from '@sapphire/framework';
+import { ClientOptions, Structures } from 'discord.js';
+import { In17nHandler, In17nOptions } from './index';
+import { In17nMessage } from './lib/extensions';
+
+Structures.extend('Message', () => In17nMessage);
+
+export class In17nPlugin extends Plugin {
+	public static [preGenericsInitialization](this: SapphireClient, options: ClientOptions): void {
+		this.in17n = new In17nHandler(options.in17n);
+
+		this.fetchLanguage = options.fetchLanguage ?? (() => options.in17n?.defaultName ?? null) ?? (() => 'en-US');
+	}
+
+	public static async [preLogin](this: SapphireClient): Promise<void> {
+		await this.in17n.init();
+	}
+}
+
+declare module 'discord.js' {
```

Comment:
This needs to be moved elsewhere, TypeScript doesn't pick augmentations in `/register`. Check the changes in the other two plugins to follow consistency.

Reply:
FYI for @kyranet : other 2 duplicate it between `index.ts` and `register.ts`. I have a hunch that just register actually works now though with `/register.d.ts` just re-exporting `dist/register.d.ts` but I'd rather not risk the biscuit.